client/lib/human-date/index.js

```diff
@@ -52,3 +49,27 @@ export default function humanDate(
 
 	return dateOrMoment.format( dateFormat );
 }
+
+export function useHumanDate( dateOrMoment, dateFormat, interval = EVERY_TEN_SECONDS ) {
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+	const [ humanDate, setHumanDate ] = useState(
+		getHumanDateString( dateOrMoment, dateFormat, moment, translate )
+	);
```

Comment:
The default value of the state is used only on initial render, but here you're calling `getHumanDateString` on every render. It will be thrown away on all renders other than first. Can be optimized as:
```js
useState( () => getHumanDateString( ... ) );
```
That answers your question about `getHumanDateString()` being called 4 times on every interval. One of them is happening here.